lib/devise_security_extension/hooks/session_non_transferable.rb

```diff
@@ -0,0 +1,33 @@
+# This hook requires session_limitable 
+
+# After each sign in, native Devise updates current_sign_in_ip and this hook
+# updates unique_session_client.
+# This is only triggered when the user is explicitly set (with set_user)
+# and on authentication. Retrieving the user from session (:fetch) does
+# not trigger it.
+Warden::Manager.after_set_user :except => :fetch do |record, warden, options|
+  if record.respond_to?(:update_current_user_agent!) && warden.authenticated?(options[:scope])
```

Comment:
to be honest I'm just fallowing how the gem was originally written, I don't mind to rewrite it so that it is by best practices but I don't want to introduce inconsistency 

my opinion is that It's better to have consistent questionable code rather than inconsistent multiple-ways-to-do-stuff kind of code :)

 please have a look at `/lib/devise_security_extension/hooks/session_limitable.rb` I'm basically just fallowing how this extension was written